library/ssl_srv.c

```diff
@@ -776,6 +818,24 @@ static int ssl_ciphersuite_match( mbedtls_ssl_context *ssl, int suite_id,
     }
 #endif
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2) && \
+    defined(MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED)
+    /* If the ciphersuite requires signing, check whether
+     * a suitable hash algorithm is present. */
+    if( ssl->minor_ver == MBEDTLS_SSL_MINOR_VERSION_3                     &&
+        ( sig_type = mbedtls_ssl_get_ciphersuite_sig_alg( suite_info ) )
```

Comment:
`if ((ret = f()) != 0)` is one thing, adding a conditional operator into the mix is another, it makes it really hard to notice the assignment. Plus we mostly do this for a return status variable, not for more complex state changes. There's nothing wrong with nesting conditionals.